app/models/shelter.rb

```diff
@@ -31,4 +34,16 @@ def alphabetical_pets
   def shelter_pets_filtered_by_age(age_filter)
     adoptable_pets.where("age >= ?", age_filter)
   end
+
+  def self.reverse_alpha_order
+    # Shelter.order(name: :desc).to_sql
+    find_by_sql("SELECT * FROM shelters ORDER BY name desc")
+  end
+
+  def self.pending_applications
+    Shelter.joins(pets: {application_pets: :application}).where(applications: {status: "Pending"}).to_a
+    
+    # joins(:applications).where("status ='Pending'").pluck(:name).count > 0 # = shelters collection
+    #select columns from different tables I want
+  end
```

Comment:
For this method, in your video, you talked about wanting to make this an instance method but this wouldn't be possible since you want to look through all of the Shelters or at least a collection of shelters and find the ones with pending applications. Since this is not finding whether just a single shelter has pending applications and doing this across all shelters, this should be a class method. 

Now for how to make this a boolean method. Since you will need to iterate over the list of shelters that come back from this query I don't think making it a boolean method provides any benefit. 

Here is another way this query could have been written.

```ruby
def self.with_pending_applications
  Shelter.joins(:application).where(applications: {status: "Pending"})
end
```

Note: 

I am able to do `Shelter.joins(:applications)` because you have set that association up already in the model. In order for `Shelter` to get to `applications` it has to go through `application_pets` which then has to go through `pets` so all the tables that you intended to join get joined with that one line. 

We need to do this `(applications: {status: "Pending"})` because both the `applications` and `application_pets` tables have a status column and we need to specify which one we're talking about. If they both didn't have a status column, we could just do `.where(status: "Pending")`.

You also don't need the `to_a` at the end.